test/option_parser.c

```diff
@@ -166,6 +166,10 @@ TEST test_string_to_int_invalid(void)
         s.type = TYPE_INT;
         s.name = "test_int";
 
+        // It should be safe to ignore this specific warning here as PASS is in fact
+        // GREATEST_PASSm(NULL) which means that greatest_info.msg is set to
+        // NULL and thus does not point to the local buf anymore.
+#pragma GCC diagnostic ignored "-Wdangling-pointer"
```

Comment:
AFAICT this warning is not really correct here. It warns about the local `buf` being set to `greatest_info.msg` which theoretically could lead to that dangling pointer, but `PASS()` at the end of the function actually resets that field to `NULL`.

Reply:
I think it's cleaner to make buf a static char. This way the storage will also stay active when the function is finished.

Reply:
I don't know if you took a look at the changes I've made after my comment. Basically, I implemented my own proposal of moving the actual test to another function and converting the top-level function into a invasive, though this turned out to be a bit invasive.

> I think it's cleaner to make buf a static char.

What do you mean by this? `static` in the scope of the test or (as `buf` is used in a couple of tests and they don't run concurrently) as a global variable?

Reply:
Every time `char buf[50]` is used, `static char buf[50]` should be used, still local to the function itself not as a global variable. This makes sure the pointer is not destroyed when the function goes out of scope, so it can be used by the testing library for messages.

Reply:
Yes, that's a bit easier than my approach. Adjusted.